pulp_docker/app/registry.py

```diff
@@ -228,4 +239,153 @@ class Registry:
                                                              artifact.file.name),
                                                 headers)
             else:
-                raise ArtifactNotFound(path)
+                return await self._stream_content_artifact(request, web.StreamResponse(), ca)
+
+    async def _stream_content_artifact(self, request, response, content_artifact):
+        """
+        Stream and optionally save a ContentArtifact by requesting it using the associated remote.
+
+        If a fatal download failure occurs while downloading and there are additional
+        :class:`~pulpcore.plugin.models.RemoteArtifact` objects associated with the
+        :class:`~pulpcore.plugin.models.ContentArtifact` they will also be tried. If all
+        :class:`~pulpcore.plugin.models.RemoteArtifact` downloads raise exceptions, an HTTP 502
+        error is returned to the client.
+
+        Args:
+            request(:class:`~aiohttp.web.Request`): The request to prepare a response for.
+            response (:class:`~aiohttp.web.StreamResponse`): The response to stream data to.
+            content_artifact (:class:`~pulpcore.plugin.models.ContentArtifact`): The ContentArtifact
+                to fetch and then stream back to the client
+
+        Raises:
+            :class:`~aiohttp.web.HTTPNotFound` when no
+                :class:`~pulpcore.plugin.models.RemoteArtifact` objects associated with the
+                :class:`~pulpcore.plugin.models.ContentArtifact` returned the binary data needed for
+                the client.
+
+        """
+        for remote_artifact in content_artifact.remoteartifact_set.all():
+            try:
+                response = await self._stream_remote_artifact(request, response, remote_artifact)
+
+            except ClientResponseError:
```

Comment:
A comment might be convenient. It can be figured out from the docstring though, if you prefer to leave as is.